jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/TaskCommon.java

```diff
@@ -38,20 +39,20 @@
 class TaskCommon {
 
   @Nullable
-  static War getWarTask(Project project) {
+  static TaskProvider<Task> getWarProviderTask(Project project) {
 
     if (!project.getPlugins().hasPlugin(WarPlugin.class)) {
       return null;
     }
 
-    if (project.getPlugins().hasPlugin("org.springframework.boot")) {
-      Task bootWar = project.getTasks().findByName("bootWar");
-      if (bootWar != null) { // Spring Boot > 2.0
-        return (War) bootWar;
+    try {
+      if (project.getPlugins().hasPlugin("org.springframework.boot")) {
+        return project.getTasks().named("bootWar");
       }
+      return project.getTasks().named(WarPlugin.WAR_TASK_NAME);
+    } catch (UnknownTaskException ignored) {
```

Comment:
I think the order is wrong here? catch should happen somewhere else? Or if we never intended to come here we should throw an illegal state exception or something.

Since we already checked if the `war-plugin` exists on the project, we should by default return `war` task.

Reply:
Ah, nice catch.

Reply:
I just put the `named('war')` inside the try..catch block to make sure I always handle the exception it throws when the task is not there. 

In this case it should never be triggered _unless_ the war task name is changed so I wanted to be safe.

Should I move it outside of the try..catch block?

Reply:
If `named("bootWar")` can throws the exception, which I think is a real possibility, this method should return `named("war")` instead of null, shouldn't it?

Reply:
oh I see what you mean.. in case that the project has `org.springframework.boot` plugin but not `bootWar` task then it will return null while there is a possibility that there is a `war` task as well (because of the warplugin check above). I will move the return outside the try..catch and remove the null return.